### PR TITLE
[hail] logger comes from Backend

### PR DESCRIPTION
**Pull request by danking**
This fixes `balding_nichols_model` in the service.

I also fixed warn => warning. Warn is deprecated in Python's logging library.

### CODE REVIEW

File: hail/python/hail/hail_logging.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    @abc.abstractmethod
+    def info(self, msg):
+        pass
+
+
+class PythonOnlyLogger:
+    def __init__(self):
+        self.logger = logging.getLogger().getChild("hail")
+        self.logger.setLevel(logging.INFO)
```
Comment (cseed) [inline hail/python/hail/hail_logging.py:24]: Why not `logging.getLogger("hail")`?

File: hail/python/hail/hail_logging.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        self.logger.error(msg)
+
+    def warning(self, msg):
+        self.logger.warning(msg)
+
+    def info(self, msg):
+        self.logger.info(msg)
+
+
+class Log4jLogger:
```
Comment (cseed) [inline hail/python/hail/hail_logging.py:36]: This can't go  here.  Java-y things can only go in spark_backend now.

File: hail/python/hail/hail_logging.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    log_pkg = None
+
+    @staticmethod
+    def get():
+        if Log4jLogger.log_pkg is None:
+            Log4jLogger.log_pkg = Env.jutils()
```
Comment (cseed) [inline hail/python/hail/hail_logging.py:42]: jutils is about to go away: when you construct this, pass it the package.  The SparkBackend will have to pass it in.

**Comment by danking** [comment]
bump

File: hail/python/hail/experimental/import_gtf.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 import hail as hl
 from hail.genetics.reference_genome import reference_genome_type
 from hail.typecheck import typecheck, nullable, sequenceof
-from hail.utils.java import info
+from hail.utils.java import Env
```
Comment (cseed) [inline hail/python/hail/experimental/import_gtf.py:6]: I think this import is unused.

File: hail/python/hail/hail_logging.py (hunk 1 above)
Comment (cseed) [reply hail/python/hail/hail_logging.py:22]: You didn't answer this.  I'm always slightly confused about how logging configuration is supposed to work.  I think this code should be something like:

```
logging.basicConfig(level=logging.INFO)
self.logger = logging.getLogger("hail")
```

?

Comment (cseed) [reply hail/python/hail/hail_logging.py:22]: I did a little digging.  `logging.getLogger().getChild("hail")` should be the same as `logging.getLogger("hail")`.

I don't think your configuration will generate output:

```
>>> import logging
>>> log = logging.getLogger('hail')
>>> log.setLevel(logging.INFO)
>>> log.info('moo')
>>>
```

Your code only turns on INFO for the hail logger. I  think I agree with that.  So I suggest this:

```
logging.basicConfig()
self.logger = logging.getLogger("hail")
self.logger.setLevel(logging.INFO)
```

Verified works:

```
>>> import logging
>>> logging.basicConfig()
>>> log = logging.getLogger('hail')
>>> log.warning('moo')
WARNING:hail:moo
>>> log.info('moo')
>>> log.setLevel(logging.INFO)
>>> log.info('moo')
INFO:hail:moo
>>> 
```

Comment (danking) [reply hail/python/hail/hail_logging.py:22]: Ah, whoops, I forgot to push the fix to use `getLogger("hail")`.

Regarding `basicConfig`, I'm pretty sure configuring that in a library is considered rude.

I do see that without basic config we don't get info statements.

```
>>> import logging
>>> x = logging.getLogger("hail")
>>> x.setLevel(logging.INFO)
>>> x.info("hi")
>>> x.warning("hi")
hi
```

Apparently when there's no handler configured, it defaults to a special handler that only prints warning and above to stderr. https://docs.python.org/3.7/howto/logging.html#what-happens-if-no-configuration-is-provided

The section on configuring logging for a library explicitly says that we shouldn't set handlers, which I believe is what basicConfig is doing. https://docs.python.org/3.7/howto/logging.html#configuring-logging-for-a-library

Ideally we would only configure if the user has not configured it, right? I guess this is hard to do because doing nothing is implicitly a configuration.

I added `basicConfig` to `context.py` but protected by the skip_logging_configuration flag.

File: hail/python/hail/backend/service_backend.py [hunk 5]
```diff
@@ -21,6 +22,13 @@ def __init__(self, deploy_config=None):
         self.url = deploy_config.base_url('query')
         self.headers = service_auth_headers(deploy_config, 'query')
         self._fs = None
+        self._logger = None
+
+    @property
+    def logger(self):
+        if self._logger is None:
+            self._logger = PythonOnlyLogger()
+        return self._logger
```
Comment (cseed) [inline hail/python/hail/backend/service_backend.py:31]: Sorry we keep going back and forth on this.  I think:
 - you should pass skip_logging_configuration from init(_service) to the backend (ask the Spark backend does)
 - the logger construction shouldn't be lazy

Then you don't need a dummy configure on the Spark backend which is misleading because the configuration happens elsewhere.

File: hail/python/hail/hail_logging.py (hunk 1 above)
Comment (cseed) [reply hail/python/hail/hail_logging.py:26]: > The section on configuring logging for a library explicitly says that we shouldn't set handlers, which I believe is what basicConfig is doing.
 > Ideally we would only configure if the user has not configured it, right? I guess this is hard to do because doing nothing is implicitly a configuration.

`basicConfig` also includes a default format, at least:

```
$ python3
>>> import logging
>>> log = logging.getLogger('hail')
>>> log.error('foo')
foo
>>> logging.basicConfig()
>>> log.error('foo')
ERROR:hail:foo
>>> 
```

No.  Libraries should not configure.  Ideally, we wouldn't configure, and we wouldn't use logging to communicate with the user.

File: hail/python/hail/backend/service_backend.py (hunk 5 above)
Comment (danking) [reply hail/python/hail/backend/service_backend.py:31]: Good point. Will fix.

File: ci/create_database.py [hunk 6]
```diff
@@ -60,8 +60,11 @@ async def write_user_config(namespace, database_name, user, config):
     from_files = ' '.join(f'--from-file={f}' for f in files)
     await check_shell(
         f'''
-kubectl -n {shq(namespace)} delete --ignore-not-found=true secret {shq(secret_name)}
-kubectl -n {shq(namespace)} create secret generic {shq(secret_name)} {from_files}
+kubectl -n {shq(namespace)} create secret generic \
+        {shq(secret_name)} \
+        {from_files} \
+        --dry-run=true -o yaml \
```
Comment (cseed) [inline ci/create_database.py:66]: What's this about?

File: ci/create_database.py [hunk 7]
```diff
@@ -82,8 +85,8 @@ async def create_database():
 
     scope = create_database_config['scope']
     _name = create_database_config['_name']
-    admin_username = create_database_config['admin_username']
-    user_username = create_database_config['user_username']
+    admin_username = create_database_config['admin_username'] + '_' + secrets.token_urlsafe(4)
+    user_username = create_database_config['user_username'] + '_' + secrets.token_urlsafe(4)
```
Comment (cseed) [inline ci/create_database.py:89]: Did some unrelated changes sneak in here?

**Comment by danking** [comment]
So the situation I was in is this:
```
* rever bad commit (hail/master)
| * my PR (danking/warnings)
|/
* bad commit
* some master commit
```

And for some reason GitHub showed the diff as including `bad commit`'s changes. Rebasing fixed the diff shown in GitHub.
